### ISSUE TITLE
Allow ContentEventJob to receive either an ActiveFedora object or a Valkyrie resource

### ISSUE THREAD

**Issue by randalldfloyd** [samvera/hyrax#4193]
### Descriptive summary
Convert ContentEventJob to allow it to receive either an ActiveFedora based object or a Valkyrie::Resource equivalent object.

### Rationale
Allow jobs to execute correctly with existing AF implementation or the Wings Valkyrie implementation.

### Expected behavior
ContentEventJob receives a Valkyrie::Resource and performs the action.

### Actual behavior
ContentEventJob raises an exception when passed a Valkyrie::Resource.

### Approach for Conversion
[See Pattern for Jobs Receiving Parameter that may be an AF Object or a Valkyrie Resource](https://docs.google.com/document/d/1f9yDjtkXapFf1DYLyPhprnB4CPGaZw9p_k8T_tcor-U/edit#heading=h.qt9rxydfd727)

### Related work

These have all be modified to support AF object or Valkyrie resource using the [established pattern](https://docs.google.com/document/d/1f9yDjtkXapFf1DYLyPhprnB4CPGaZw9p_k8T_tcor-U/edit#heading=h.qt9rxydfd727). 

* Issue #4101 Allow GrantEditToMembersJob to receive either an AF object or a Valkyrie resource as the Work (PR #4123)
* Issue #4102 Allow GrantReadToMembersJob to receive either an AF object or a Valkyrie resource as the Work (PR #4123)
* Issue #4103 Allow RevokeReadToMembersJob to receive either an AF object or a Valkyrie resource as the Work (PR #4123)
* Issue #4099 Add GlobalID::Identification for resources passed to jobs (PR #4124)

**Comment by jeremyf** [comment]
The ContentEventJob has 6 subclasses:

- [ ] ContentUpdateEventJob
- [ ] ContentNewVersionEventJob
- [ ] FileSetAttachedEventJob
- [ ] ContentRestoredVersionEventJob
- [ ] ContentDepositorChangeEventJob
- [ ] ContentDepositEventJob

This requires checking their implementations, but I suspect they are almost entirely a "noop"

**Comment by jeremyf** [comment]
The `Hyrax::WithEvents#log_events` ([see code](https://github.com/samvera/hyrax/blob/a8597884e1ee1802119ee9e82cb7ae9e52ee203f/app/models/concerns/hyrax/with_events.rb#L25-L27)) exposes the common logging method.

Below are the five places that mixin the `Hyrax::WithEvents` module:

* app/models/concerns/hyrax/user.rb
* app/models/concerns/hyrax/file_set_behavior.rb
* app/models/hyrax/resource.rb
* app/models/concerns/hyrax/work_behavior.rb
* app/presenters/hyrax/file_set_presenter.rb

In other words a `Hyrax::Resource` (which is a `Valkyrie::Resource`)
and those that use `Hyrax::WorkBehavior` (e.g., `GenericWork`) will
each have the necessary `log_events` method.  Which is the most common pattern of the event_log and it's progeny.

With that information, I am comfortable closing this issue.
